src/Sulu/Bundle/WebsiteBundle/Routing/PortalLoader.php

```diff
@@ -22,33 +23,45 @@
  */
 class PortalLoader extends Loader
 {
+    /**
+     * @var WebspaceManagerInterface
+     */
+    private $webspaceManager;
+
+    public function __construct(WebspaceManagerInterface $webspaceManager)
+    {
+        $this->webspaceManager = $webspaceManager;
+    }
+
     public function load($resource, $type = null)
     {
-        $collection = new RouteCollection();
+        $prefixes = [];
+        foreach ($this->webspaceManager->getPortalInformations() as $portalInformation) {
+            // cast null to string as prefix can be empty string
+            $prefixes[] = preg_quote((string) $portalInformation->getPrefix());
+        }
+
+        // need to omit prefix from path if it must be empty to pass symfony route validation
+        $prefixPattern = implode('|', array_unique($prefixes));
+        $pathPrefix = empty($prefixPattern) ? '' : '{prefix}';
+        $requirements = empty($prefixPattern) ? [] : ['prefix' => $prefixPattern];
```

Comment:
prefix need always exist else it will break things.

Reply:
when would this happen? 🤔 
if there is a single webspace that uses `{host}` as url, the prefix requirement will be an empty string. this causes the following error:
```
Routing requirement for "prefix" cannot be empty in @SuluSearchBundle/Resources/config/routing_website.yml
```

Reply:
Seems like then another BC Break then because at current state with empty resourceLocatorPrefix url generation works correctly:

```yaml
{{ path('sulu_community.password_forget', { prefix: request.resourceLocatorPrefix }) }}
```

or 

```yaml
{{ path('sulu_community.password_forget', { prefix: '' }) }}
```

As said we can not remove the `prefix` as this would be a bc break which accidentally adding a query parameter which should not be there. The route path should not change because I add in future another webspace or change the webspace configuration. We should just edit the current regex that it allow only the configured prefixes.

Reply:
its not about the generation - the problem is that symfony does not accept a route with an empty requirement.. in the case described above, the regex (`$prefixPattern`) would be an empty string. this leads to the exception

Reply:
Think we will find a regex which will match this and symfony doesn't throw an error.

Reply:
The following should work:

```
static:
    path: /{prefix}static-route
    controller: App\Controller\DefaultController::index
    requirements:
        prefix: '(^$)?|test/'
```

So the prefix for empty resourceLocatorPrefix should be `(^$)?`.

Reply:
you are right 👍  i figured that `()` works too and updated the PR